common/src/api/internal/shared.rs

```diff
@@ -422,3 +422,136 @@ impl fmt::Display for PortFec {
         }
     }
 }
+
+/// Description of source IPs allowed to reach rack services.
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case", tag = "allow", content = "ips")]
+pub enum AllowedSourceIps {
+    /// Allow traffic from any external IP address.
+    Any,
+    /// Restrict access to a specific set of source IP addresses or subnets.
```

Comment:
Does this really allow for IP addresses or subnets? Our JSON Schema at least for IpNet doesn't allow a "/" to be omitted.

Reply:
Yeah, that's the problem. I think we'd need an enum of `IpAddr` or `IpNet` to support both directly. I'm not sure how to do that yet without carrying that variant around everywhere in the code.

Reply:
We've deferred support an address or a network, everything is just the latter at this point. We might include support for both in the future if we need it, but it's a lot of drudgery in the OpenAPI definitions for minimal gain at this point.